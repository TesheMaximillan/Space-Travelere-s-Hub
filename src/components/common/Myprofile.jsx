import React from 'react';
import MissionProfile from './MissionProfile';
import RocketProfile from './RocketProfile';
import styles from './Myprofile.module.scss';

function Myprofile() {
  const {
    profileContainer, profiles, profile, list,
  } = styles;
  return (
    <div className={profileContainer}>
      <div className={profiles}>
        <h1>My Missions</h1>
        <ul className={profile}>
          <MissionProfile list={list} />
        </ul>
      </div>
      <div className={profiles}>
        <h1>My Rockets</h1>
        <ul className={profile}>
          <RocketProfile list={list} />
        </ul>
      </div>
    </div>
  );
}

export default Myprofile;
