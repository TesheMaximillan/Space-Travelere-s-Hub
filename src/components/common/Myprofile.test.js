import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Myprofile from './Myprofile';
import renderContext from '../storeMock';

describe('Myprofile component', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(renderContext(<Myprofile />));
  });

  it('should render with given state from redux store', () => {
    // Assert
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render Myprofile', () => {
    // Arrange
    render(renderContext(<Myprofile />));

    // Assert
    expect(screen.getByText(/Missions/i)).toBeInTheDocument();
  });
});
