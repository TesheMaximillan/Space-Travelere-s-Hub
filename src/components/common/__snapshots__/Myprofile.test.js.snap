// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Myprofile component should render with given state from redux store 1`] = `
<div
  className="profileContainer"
>
  <div
    className="profiles"
  >
    <h1>
      My Missions
    </h1>
    <ul
      className="profile"
    />
  </div>
  <div
    className="profiles"
  >
    <h1>
      My Rockets
    </h1>
    <ul
      className="profile"
    />
  </div>
</div>
`;
